apps/api/src/rest/routes/programs.ts

```diff
@@ -0,0 +1,248 @@
+import { defaultHook } from "$hooks";
+import { productionCache } from "$middleware";
+import {
+  errorSchema,
+  majorRequirementsQuerySchema,
+  majorRequirementsResponseSchema,
+  majorsResponseSchema,
+  minorRequirementsQuerySchema,
+  minorRequirementsResponseSchema,
+  minorsResponseSchema,
+  programRequirementSchema,
+  responseSchema,
+  specializationRequirementsQuerySchema,
+  specializationRequirementsResponseSchema,
+  specializationsResponseSchema,
+} from "$schema";
+import { ProgramsService } from "$services";
+import { OpenAPIHono, createRoute } from "@hono/zod-openapi";
+import { database } from "@packages/db";
+
+const programsRouter = new OpenAPIHono<{ Bindings: Env }>({ defaultHook });
+
+programsRouter.openAPIRegistry.register("programRequirement", programRequirementSchema);
+
+const majorsRoute = createRoute({
+  summary: "Retrieve majors",
+  operationId: "majorsRoute",
+  tags: ["Programs"],
+  method: "get",
+  path: "/majors",
+  description: "List all available majors in UCI's current catalog.",
```

Comment:
Nit: UCI refers to the resource that outlines all programs that it offers as a "catalogue", and not the `en-US` spelling of "catalog".

This nit applies to all uses of "catalog" in this PR.